statics/js/topologymanager_package/src/topologymanager/data_source/base_skydive_data_source.ts

```diff
@@ -0,0 +1,56 @@
+import DataSourceI from "./interface";
+import * as events from 'events';
+import WsHandler from './websocket';
+
+export default class BaseSkydiveDataSource implements DataSourceI {
+    sourceType: string = "skydive";
+    dataSourceName: string = "base_skydive_data_source";
+    e: events.EventEmitter = new events.EventEmitter();
```

Comment:
> Relying on events and not callback makes to call harder to understand and to debug.
> 
> Or at least you could defines const that could be used.

I agree about that but it's a point for next refactoring. It would be small changes comparing to the current changeset size.

> There was already a subscriber mechanism, I do not see the point to use this EventEmitter
> https://github.com/skydive-project/skydive/blob/master/statics/js/websocket.js#L62
> 
> Other components could already subscribe to the graph to do something else, ex:
> https://github.com/skydive-project/skydive/blob/master/statics/js/components/rule-detail.js#L750

Such subscription mechanism breaks the one responsibilty SOLID principle. Each object which needs to have such mechanism needs to implement it on its own but it makes the code hard to reuse, etc
that's why we have an object EventEmitter in nodejs and it works fine in browser, and basically it is de-facto standart for nodejs.